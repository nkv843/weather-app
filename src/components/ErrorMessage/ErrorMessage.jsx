import React from 'react';
import PropTypes from 'prop-types';
import classNames from './ErrorMessage.module.css';

const ErrorMessage = ({ serverError }) => (
  serverError
    ? (
      <div className={classNames.message}>
        <span className={classNames.messageFrame}>Oops, something went wrong ü•≤</span>
        <br />
        <span className={classNames.messageFrame}>‚öíÔ∏è</span>
        {serverError}
      </div>
    )
    : null
);

ErrorMessage.propTypes = {
  serverError: PropTypes.string,
};

ErrorMessage.defaultProps = {
  serverError: null,
};

export default ErrorMessage;
