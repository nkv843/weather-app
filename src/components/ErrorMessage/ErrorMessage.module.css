.message {
  margin: 20px 50px;
  color: rgb(65, 45, 88);
  font-size: 22px;
  font-weight: 500;
}

.messageFrame {
  font-size: 23px;
}
