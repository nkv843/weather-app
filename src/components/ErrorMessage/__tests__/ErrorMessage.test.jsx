import React from 'react';
import ErrorMessage from '../ErrorMessage';
import renderWithProviders from '../../../common/utils/customRender';

describe('ErrorMessage', () => {
  it('should match snapshot', () => {
    const { baseElement } = renderWithProviders(<ErrorMessage serverError="Error: oops" />);
    expect(baseElement).toMatchSnapshot();
  });
});
