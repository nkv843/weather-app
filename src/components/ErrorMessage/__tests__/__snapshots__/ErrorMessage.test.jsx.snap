// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should match snapshot 1`] = `
<body>
  <div>
    <div
      class="message"
    >
      <span>
        Oops, something went wrong ü•≤
      </span>
      <br />
      <span>
        ‚öíÔ∏è
      </span>
      Error: oops
    </div>
  </div>
</body>
`;
